tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/PortalServerLaunchConfigDelegate.java

```diff
@@ -132,8 +136,30 @@ public void serverChanged( ServerEvent event )
                         LiferayServerCore.logError( "Could not reinitialize source lookup director", e );
                     }
                 }
+                else if((event.getKind() & ServerEvent.SERVER_CHANGE)>0 && event.getState() == IServer.STATE_STARTED) {
+					IModule[] modules = server.getModules();
+
+					for (IModule module : modules) {
+						IBundleProject project = LiferayCore.create(IBundleProject.class, module.getProject());
+						if (project != null && project instanceof IWatchableProject
```

Comment:
So instead of having a `supported` just do this
```IWatchableProject watchableProject = LiferayCore.create(IWatchableProject.class)```
and then you don't need to do a `supported`